### ISSUE TITLE
Implications when used by custom loss functions

### ISSUE THREAD

**Issue by ogencoglu** [superlinear-ai/conformal-tights#36]
Are intervals valid for arbitrary loss functions? Let's say a multi-task problem when some subset of data is penalized by squared loss and some other subset is an arbitrary classification loss. Total loss that is minimized by the model is a weighted sum. What are the implications on the intervals?

**Comment by lsorber** [comment]
The prediction intervals (and predicted quantiles) are independent of your choice of loss function.

In Conformal Tights, the loss function you pick (e.g., 'l1', 'l2', or some combination) only affects the point predictions you make with the regressor you pick (e.g., an `XGBRegressor` or an `LGBMRegressor`).

The prediction intervals and predicted quantiles are constructed with quantile regression using the quantile loss. More specifically, we use XGBoot's vector quantile regression to obtain a set of (noncoherent) nonconformity estimates, and then we use coherent quantile regression to produce calibrated and coherent quantile predictions.

**Comment by ogencoglu** [comment]
Thanks for the explanation. What is the difference of this method vs training several models with different quantile regression losses in the first place instead of the 2-stage approach?

**Comment by lsorber** [comment]
If you were to directly apply quantile regression on the quantiles you want to predict on, there is no guarantee that those quantiles won't [cross](https://github.com/dmlc/xgboost/issues/9848) [each other](https://github.com/microsoft/LightGBM/issues/3447). For example, the 25th percentile prediction may be larger than the 75th percentile prediction.

The two-stage approach allows us to ensure that the quantiles are coherent (i.e., that they don't cross each other) and that they're conformally calibrated.
